### PR TITLE
Bugfix/EH-233 Content-Length mismatch

### PR DESCRIPTION
**Pull request by oleg1std**
- Fix Content-Length mismatch and ProblemDetails deserialization in the ErrorHandling middleware
- Add ProblemDetails extensions handling to the factory

### CODE REVIEW

File: HappyTravel.ErrorHandling/ErrorHandlingMiddleware.cs [hunk 1]
```diff
@@ -55,15 +53,34 @@ public async Task Invoke(HttpContext httpContext, PublicProblemDetailsFactory fa
                     Detail = ex.Message,
                     Status = (int) HttpStatusCode.InternalServerError
                 });
-                await StreamHelper.CopyDetailsTo(originalBody, problemDetails);
```
Comment (kirillta) [inline HappyTravel.ErrorHandling/ErrorHandlingMiddleware.cs:58]: Looks like `CopyDetailsTo` becomes unused

File: HappyTravel.ErrorHandling/ErrorHandlingMiddleware.cs [hunk 2]
```diff
@@ -55,15 +53,34 @@ public async Task Invoke(HttpContext httpContext, PublicProblemDetailsFactory fa
...
+                
+                await StreamHelper.Rewrite(proxyStream, problemDetails);
```
Comment (kirillta) [inline HappyTravel.ErrorHandling/ErrorHandlingMiddleware.cs:57]: The name `Rewrite` isn't clear in the context. Are we rewrite problem details of the stream or one stream to another?

**Review by mingazhev — APPROVED** [review]
With @kirillta  comments.
And many thanks for fixing that

File: HappyTravel.ErrorHandling/ErrorHandlingMiddleware.cs (hunk 2 above)
Comment (oleg1std) [reply HappyTravel.ErrorHandling/ErrorHandlingMiddleware.cs:57]: The method has been renamed to RenameDetails

File: HappyTravel.ErrorHandling/ErrorHandlingMiddleware.cs (hunk 1 above)
Comment (oleg1std) [reply HappyTravel.ErrorHandling/ErrorHandlingMiddleware.cs:58]: This method is used in UseProblemDetailsExceptionHandler

**Review by kirillta — APPROVED** [review]

**Pull request merged**